src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentNodeWriter.cs

```diff
@@ -340,6 +333,40 @@ protected void WriteComponentTypeInferenceMethod(CodeRenderingContext context, C
         }
     }
 
+    protected static void WriteComponentAttributeName(CodeRenderingContext context, ComponentAttributeIntermediateNode attribute, bool allowNameof = true)
+    {
+        if (allowNameof && attribute.BoundAttribute?.ContainingType is string containingType)
+        {
+            containingType = attribute.Annotations[ComponentMetadata.Component.ConcreteContainingType] as string ?? containingType;
+
+            // nameof(containingType.PropertyName)
+            context.CodeWriter.Write("nameof(");
+            TypeNameHelper.WriteGloballyQualifiedName(context.CodeWriter, containingType);
+            context.CodeWriter.Write(".");
```

Comment:
What if the type is in the global namespace?

Reply:
This is broken already today :( 

https://github.com/dotnet/razor/issues/10609

Reply:
But I think it should work right? The attribute must always be on some kind of type, and while the type itself might not have a namespace we're not constructing that part, that's handled by roslyn for us.

Reply:
Yeah, so the issue is actually only with `@bind-` syntax. Components in top level namespaces work otherwise. Confirmed it's not an issue here and added a test to cover it.